master/internal/kubernetes/spec.go

```diff
@@ -175,8 +176,8 @@ func (p *pod) configureVolumes(
 }
 
 func (p *pod) modifyPodSpec(newPod *k8sV1.Pod, scheduler string) {
-	if scheduler != "coscheduler" || newPod.Spec.SchedulerName != "" ||
-		p.taskSpec.Description() == "cmd" {
+	if scheduler != coscheduler || p.taskSpec.Description() == "cmd" ||
+		(newPod.Spec.SchedulerName != "" && newPod.Spec.SchedulerName != coscheduler) {
```

Comment:
Nit: cmd is another magic string we should try to factor out of here and  task_spec.go.

Also - why is preemption not included here?

Reply:
will change cmd. Since preemption and coscheduling are the same on determined's side, I just had the preemption option basically just point to the coscheduler. Should I make it more verbose?

Reply:
Oh I had missed that. It's not new on that line anyway... I feel like we need distinct terms for coscheduler, coscheduler-with-preemption, and the combination. But that can be a separate discussion.

Reply:
Since we're introducing another scheduler, I'm going to refactor this part of the code to be more friendly to different scheduler types.

Reply:
Also, `Description()` will always return "cmd", "gc", or an experiment name.